src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -436,6 +436,77 @@ private async Task StopAsyncCore(bool disposing)
             }
         }
 
+#if NETCOREAPP3_0
+        public async IAsyncEnumerable<object> StreamAsyncCore(string methodName, Type returnType, object[] args, CancellationToken cancellationToken = default)
```

Comment:
What if we just did
```c#
return await StreamAsChannelCoreAsync(methodName, returnType, args, cancellationToken).ReadAllAsync();
```

Reply:
What we should actually be doing here is using the `MakeCancelableAsyncEnumerableFromChannel<T>` method on the `AsyncEnumerableAdapters` class

Reply:
Sounds good. We can just share the source for that file for now.